automation/sheet_checker.py

```diff
@@ -0,0 +1,165 @@
+
+
+    def is_sheet_updated(self):
+        sheet_last_updated_ts = dateutil.parser.parse(self.get_sheet_update_ts())
+        last_run_ts = self.fetch_last_run_ts()
+        if not last_run_ts:
+            return False
+        else:
+            job_last_run_ts = dateutil.parser.parse(last_run_ts)
+            job_last_run_ts = job_last_run_ts.replace(tzinfo=pytz.UTC)
+            return sheet_last_updated_ts >= job_last_run_ts
+
+
+    def get_sheet_update_ts(self):
+        gc = pygsheets.authorize(service_file='client_secret.json')
+        gc.enableTeamDriveSupport = True 
+        sheet = gc.open_by_key(self.sheet_key)
+        last_updated_ts = sheet.updated
+        return last_updated_ts
+
+    def update_last_ran_ts(self):
+        timestamp = datetime.strftime(datetime.now(), '%Y-%m-%d %H:%M:%S')
+        buff = BytesIO()
+        buff.write(timestamp.encode('utf-8'))
+        buff.seek(0)
+        self.s3_client.put_object(Body=buff, Bucket='tji-timestamps', Key=self.dataset)
+        self.logger.info("Updated %s timestamp." % self.dataset)
+
+    def fetch_last_run_ts(self):
+        try:
+            timestamp = self.s3_client.get_object(Bucket='tji-timestamps', Key=self.dataset)
+            body = timestamp['Body']
+            last_run_ts = body.read()
+            return last_run_ts
+        except ClientError as e:
+            if e.response['Error']['Code'] == "404":
+                self.logger.info("No timestamp found.")
+            else:
+                self.logger.exception(e)
+                self.logger.info("Something went wrong while fetching timestamp.")
+
+    def set_up_environment(self):
+        dataset = self.dataset.upper()
+        os.environ['CLEAN_%s_S3' % dataset] = 'TRUE'
+        os.environ['COMPRESS_%s_S3' % dataset] = 'TRUE'
+        os.environ['COMPRESS_DATASET'] = dataset
+
+    def clean_up_environment(self):
+        dataset = self.dataset.upper()
+        os.unsetenv('CLEAN_%s_S3' % dataset)
+        os.unsetenv('COMPRESS_%s_S3' % dataset)
+        os.unsetenv('COMPRESS_DATASET')
+
+class CDRChecker(SheetChecker):
+    def __init__(self, *args, **kwargs):
+        super(CDRChecker, self).__init__(*args, **kwargs)
+
+class OISChecker(SheetChecker):
+    def __init__(self, *args, **kwargs):
+        super(OISChecker, self).__init__(*args, **kwargs)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Check gsheet for changes and re-clean and re-compress.')
+    parser.add_argument('-force', action='store_true', help='Force everything to re-clean and re-compress')
+    parser.add_argument('-cdr', action='store_true', help='Enable CDR')
+    parser.add_argument('-ois', action='store_true', help='Enable OIS')
+    args = parser.parse_args()
+    force = False
+    if args.force:
+        force = True
+
+    if args.cdr:
+        cdr = CDRChecker(dataset='cdr', sheet_key='15othpOTOwB25_ccde4jHK3MaQDAvQyJSXJIVL3eihHg', cleaning_nbs=['clean_cdr.ipynb'])
```

Comment:
We realllly shouldn't hardcode the sheet keys. They could either be command-line args or environment variables (e.g. `SHEET_KEY_CDR`)

Reply:
Yeah, my bad. Same thing here, I'll pull all of these into a yaml which should be a little easier to manage in the future.